src/sage/libs/flint/qsieve.pyx

```diff
@@ -0,0 +1,69 @@
+"""
+Interface to FLINT's ``qsieve_factor()``. This used to interact
+with an external "QuadraticSieve" program, but its functionality has
+been absorbed into flint.
+"""
+
+from cysignals.signals cimport sig_check, sig_on, sig_off
+
+from sage.libs.flint.fmpz cimport fmpz_t, fmpz_init, fmpz_set_mpz, fmpz_get_mpz
+from sage.libs.flint.fmpz_factor cimport *
+from sage.libs.flint.qsieve cimport *
+from sage.libs.gmp.mpz cimport mpz_t, mpz_init, mpz_set, mpz_clear
+from sage.rings.integer cimport Integer
+
+def qsieve(n):
```

Comment:
Why not `def qsieve(Integer n)`?

And I suggest to use `factor_qsieve` rather than `qsieve`.

Reply:
`qsieve()` is in the global namespace and is documented in a few places, so I was reluctant to break too many things. The old implementation called `n = ZZ(n)` at the start, so, for example, you could pass it a rational number (with denominator 1) and it would factor it for you. I left the name alone for similar reasons.

If it were entirely up to me, I would remove `qsieve` from the global namespace, leaving only `n.factor(algorithm="qsieve")` and `factor(n, algorithm="qsieve")` as the official interfaces to it.

Reply:
This is not ideal because things like `qsieve('5')` or `qsieve(5.0)` would then work. It is fine to leave it but not a great interface.